### PR TITLE
UI: Minting success page

### PR DESCRIPTION
**Pull request by kpyszkowski**
Ref: #5

This PR introduces minting success page

image

Please note: the summary list (above the button) will be included with separate PR when required changes will get merged regarding to #36

### CODE REVIEW

File: src/pages/tBTC/Bridge/DepositDetails.tsx
Comment (michalsmiarowski) [inline src/pages/tBTC/Bridge/DepositDetails.tsx:236]: Hmm, look like we have two right sections right now:

image

I think this issue might be related to this comment: https://github.com/threshold-network/bitcoin-on-base/pull/18#discussion_r1508449344. We could combine the fix for both by setting an `undefined` for right section (probably only for depoist details page) and render the correct one inside the actual component likt it is done right now, in this PR.

File: src/pages/tBTC/Bridge/DepositDetails.tsx
Comment (michalsmiarowski) [inline src/pages/tBTC/Bridge/DepositDetails.tsx:210]: Let's use `DepositDetailsTimelineStep` type instead of writing "completed" as a plain string.

File: src/pages/tBTC/Bridge/DepositDetails.tsx
Comment (michalsmiarowski) [inline src/pages/tBTC/Bridge/DepositDetails.tsx:258]: Same as above

File: src/pages/tBTC/Bridge/DepositDetails.tsx
Comment (michalsmiarowski) [inline src/pages/tBTC/Bridge/DepositDetails.tsx:272]: Same as above

**Review by michalsmiarowski — CHANGES_REQUESTED** [review]
Left some comments to look at before the merge

File: src/pages/tBTC/Bridge/components/DepositDetailsStep.tsx
Comment (kpyszkowski) [reply src/pages/tBTC/Bridge/components/DepositDetailsStep.tsx]: Ref commit: e42a4219ac9afe97d115dc6290148b70109fa730

File: src/pages/tBTC/Bridge/DepositDetails.tsx
Comment (kpyszkowski) [reply src/pages/tBTC/Bridge/DepositDetails.tsx:236]: In this PR I focused only on Success step itself. I didn't touch anything around (like removing existing UI parts) because it's a part of Deposit Details view and it's already done in #18, good example why I'm missing Storybook here 😆 . 
About your suggestion - I elaborated on it in reply to the comment you mention.

File: src/pages/tBTC/Bridge/DepositDetails.tsx
Comment (kpyszkowski) [reply src/pages/tBTC/Bridge/DepositDetails.tsx:210]: `DepositDetailsTimelineStep` is a type definition and cannot be used here. Would you like to add corresponding enumeration structure?

File: src/pages/tBTC/Bridge/DepositDetails.tsx
Comment (kpyszkowski) [reply src/pages/tBTC/Bridge/DepositDetails.tsx:258]: Same as above

File: src/pages/tBTC/Bridge/DepositDetails.tsx
Comment (kpyszkowski) [reply src/pages/tBTC/Bridge/DepositDetails.tsx:272]: Same as above

File: src/pages/tBTC/Bridge/components/DepositDetailsStep.tsx
Comment (michalsmiarowski) [inline src/pages/tBTC/Bridge/components/DepositDetailsStep.tsx:1]: Non-blocking, but should be addressed in a separate PR:

Timeline and Transaction history looks little bit off on smaller screens IMO. We should at least give them the same margins on the left and right as for the `Token Contract` button or center it.

image

**Review by michalsmiarowski — APPROVED** [review]
Left some non-blocking comments but overall looking good

File: src/pages/tBTC/Bridge/DepositDetails.tsx
Comment (michalsmiarowski) [reply src/pages/tBTC/Bridge/DepositDetails.tsx]: No, let's not block this PR for that. I'm resolving the conversation and we can merge.

File: src/pages/tBTC/Bridge/components/DepositDetailsStep.tsx
Comment (kpyszkowski) [reply src/pages/tBTC/Bridge/components/DepositDetailsStep.tsx:1]: So [this line](https://github.com/threshold-network/bitcoin-on-base/blob/main/src/pages/tBTC/Bridge/Minting/MintingDepositTimeline.tsx#L113) should be updated?
```ts
id: DepositDetailsStep.MintingCompleted,
```

Comment (michalsmiarowski) [reply src/pages/tBTC/Bridge/components/DepositDetailsStep.tsx:1]: Nope, I think the issue is here:

https://github.com/threshold-network/bitcoin-on-base/blob/a516867a3c7318fe551e70690c2b76abe69f9eed/src/pages/tBTC/Bridge/Minting/MintingDepositTimeline.tsx#L163

`itemIdIndex` is never lower than `currentStepIndex` when it comes to last timeline item.

**Pull request merged**
